app/data_providers/translation.py

```diff
@@ -0,0 +1,21 @@
+import json
+from functools import lru_cache
+from os import path
+from typing import Dict
+
+from app.models.localemodel import LocaleModel
+
+
+class TranslationProvider:
+    def __init__(self, translations_dir: str):
+        self.translations_dir = translations_dir
+
+    @lru_cache(maxsize=None)  # Python 3.9 adds a @cache decorator for a simple unbounded function cache.
+    def get_translations(self, locale: LocaleModel, filename: str) -> Dict[str, str]:
```

Comment:
why not just load it once on startup and handle extra functionality when it's needed?